src/w2grid.js

```diff
@@ -4733,6 +4733,37 @@
 
             var gridBody = $('#grid_'+ this.name +'_body', obj.box).html(bodyHTML),
                 records = $('#grid_'+ this.name +'_records', obj.box);
+            var frecords = $('#grid_'+ this.name +'_frecords', obj.box);
+            records
+            .on('mouseover', 'tr', function(ev) {
+              $('#grid_'+ this.name +'_frec_' + w2utils.escapeId($(this).attr('recid'))).addClass('w2ui-record-hover')
+            })
+            .on('mouseout', 'tr', function(ev) {
+              $('#grid_'+ this.name +'_frec_' + w2utils.escapeId($(this).attr('recid'))).removeClass('w2ui-record-hover')
+            })
+
+            frecords
+            .on('mouseover', 'tr', function(ev) {
+              $('#grid_'+ this.name +'_rec_' + w2utils.escapeId($(this).attr('recid'))).addClass('w2ui-record-hover')
+            })
+            .on('mouseout', 'tr', function(ev) {
+              $('#grid_'+ this.name +'_rec_' + w2utils.escapeId($(this).attr('recid'))).removeClass('w2ui-record-hover')
+            })
+            var self = this;
+            if(w2utils.isIOS)
+              records.add(frecords)
+              .on('click', 'tr', function(ev) {
+                self.dblClick($(this).attr('recid'), ev);
+              })
+            else
+              records.add(frecords)
+              .on('click', 'tr', function(ev) {
+                self.click($(this).attr('recid'), ev);
+              })
+              .on('contextmenu', 'tr', function(ev) {
+                self.contextMenu($(this).attr('recid'), null, ev);
+              })
+
```

Comment:
Your code does look cleaner, but I am not sure it covers all use cases. Also, please note that due to lazy loading records might not be present, therefore, adding events with `on('click'...` might be wrong. I am not sure, though.

Reply:
when parent node has an event handler - all children use it
[Event Handler Attachment](https://api.jquery.com/on/)